### DISCUSSION TITLE
Twilio Whatsapp OTP Error - A text message body or media urls must be specified (Error Code: 21619)

### DISCUSSION THREAD

**Discussion by ChinKX** [supabase/supabase#21445] [Questions] [answered]
Hi, I am using Supabase auth with Twilio to send Whatsapp OTP. I have configured all the necessary settings on supabase dashboard (refer to 1st screenshot) but the whatsapp otp is not working (Note: The sms OTP is working perfectly!). I have also setup the whastapp template on Twilio also (refer to 3rd screenshot).

This is the error I received from Supabase auth: 400: Error sending sms OTP: A text message body or media urls must be specified. More information: https://www.twilio.com/docs/errors/21619 as shown in 2nd screenshot.

Code Snippet:
```js
const signInMobile: AuthenticationAction["signInMobile"] = async ({
  mobileNumber,
  channel,
}) => {
  const { error } = await supabase.auth.signInWithOtp({
    phone: mobileNumber,
    options: {
      channel, // 'sms' | 'whatsapp'
    },
  })
  if (error) {
    trackAuthError({
      mobile: mobileNumber,
      error,
    })
    throw new Error(
      [`Name: ${error.name}`, `Message: ${error.message}`].join("\n"),
    )
  }
  setAuthenticationState({
    status: "authenticating",
    mobileNumber,
  })
}
```

Supabase-js Sdk Version: `2.39.7`

Setup References:
- https://supabase.com/docs/guides/auth/phone-login/twilio
- https://supabase.com/docs/guides/auth/phone-login/twilio#whatsapp-otp-logins

Any help is highly appreciated! Thanks in advance!

| 1st Screenshot  |
| ------------- |
| ![image](https://github.com/supabase/supabase/assets/40538074/6b904909-49b9-4ff1-9957-28a753bff46a) |

| 2nd Screenshot  |
| ------------- |
| ![image](https://github.com/supabase/supabase/assets/40538074/dba052d5-27c6-4004-a000-c7d11653bb69) |

| 3rd Screenshot  |
| ------------- |
| ![image](https://github.com/supabase/supabase/assets/40538074/ca93ab3f-0181-4c9b-974f-4721cdf34917) |

**Comment by J0** [comment]
Hey 

Thanks for patiently waiting - It's been a while since the original post so wanted to check if the issue persists. Let us know!

Joel

**Comment by ChinKX** [comment, marked as answer]
Hi @J0, thanks for following up! We managed to solve the issue by using Twilio Verify API instead of Twilio Programmable Messaging API on Supabase dashboard. Thanks!

![image](https://github.com/supabase/supabase/assets/40538074/ccf2b61a-669f-4467-81b0-79b3266bbe3a)
